src/evaluator/matchers/semver_lte.ts

```diff
@@ -0,0 +1,14 @@
+import { ILogger } from '../../logger/types';
+import { Semver } from '../../utils/Semver';
+
+export function lessThanEqualToSemverMatcherContext(log: ILogger, ruleAttr: string) {
+  const ruleSemver = new Semver(ruleAttr);
```

Comment:
same here?

Reply:
Idem. In JS we have to minimize code size, so we try to reuse as much code as possible. 
Exception handling is in the `parser` function at `src/evaluator/parser/index.ts`.